src/components/item/publish/PublicationButton.tsx

```diff
@@ -0,0 +1,216 @@
+import LinkIcon from '@mui/icons-material/Link';
+import { LoadingButton } from '@mui/lab';
+import { Alert, Button, Stack, Typography } from '@mui/material';
+
+import { ClientHostManager, PackedItem, ShortLinkPlatform } from '@graasp/sdk';
+
+import { CheckIcon } from 'lucide-react';
+
+import { ADMIN_CONTACT } from '@/config/constants';
+import { useBuilderTranslation } from '@/config/i18n';
+import { mutations } from '@/config/queryClient';
+import { buildItemPublicationButton } from '@/config/selectors';
+import { BUILDER } from '@/langs/constants';
+import { PublicationStatus } from '@/types/publication';
+
+import ContentLoader from '../../common/ContentLoader';
+import useModal from '../../hooks/useModal';
+import usePublicationStatus from '../../hooks/usePublicationStatus';
+import PublicVisibilityModal from './PublicVisibilityModal';
+
+type Props = {
+  item: PackedItem;
+  notifyCoEditors: boolean;
+};
+
+type PublicationButtonMap = {
+  [status in PublicationStatus]: {
+    description: JSX.Element | string;
+    elements: JSX.Element | JSX.Element[];
+  };
+};
+
+const { useUnpublishItem, usePublishItem, usePostItemValidation } = mutations;
+
+export const PublicationButton = ({
+  item,
+  notifyCoEditors,
+}: Props): JSX.Element => {
+  const { t } = useBuilderTranslation();
+  const { id: itemId, public: isPublic } = item;
+  const { status, isinitialLoading: isStatusFirstLoading } =
+    usePublicationStatus({ item });
+  const { isOpen, openModal, closeModal } = useModal();
```

Comment:
```suggestion
  const { isOpen, openModal, closeModal } = useModalStatus();
```
since it does not return the modal, it only handles the open status right?

Reply:
Yes, it only allows to open or close the modal.